src/parser/mage/fire/modules/features/Pyroclasm.tsx

```diff
@@ -37,25 +43,25 @@ class Pyroclasm extends Analyzer {
   }
 
   //Gets the Begin Cast Event. This is used for determining if a cast is instant or not.
-  onPyroblastBeginCast(event) {
+  onPyroblastBeginCast(event: BeginCastEvent) {
     this.beginCastEvent = event;
   }
 
   //Gets the Cast Event. This is used for determining if a cast is instant or not.
-  onPyroblastCast(event) {
+  onPyroblastCast(event: CastEvent) {
     this.castEvent = event;
   }
 
   //Counts the number of times Pyroclasm was applied
-  onPyroclasmApplied(event) {
+  onPyroclasmApplied(event: ApplyBuffEvent | ApplyBuffStackEvent) {
     this.totalProcs += 1;
     this.buffAppliedEvent = event;
     debug && this.log("Buff Applied");
   }
 
   //Checks to see if Pyroclasm was removed because it was used (there was a non instant pyroblast within 250ms) or because it expired.
-  onPyroclasmRemoved(event) {
-    const isInstantCast = (this.castEvent.timestamp - this.beginCastEvent < CAST_BUFFER);
+  onPyroclasmRemoved(event: RemoveBuffEvent | RemoveBuffStackEvent) {
+    const isInstantCast = (this.castEvent.timestamp - this.beginCastEvent.timestamp < CAST_BUFFER);
```

Comment:
```suggestion
    const channelingTime = this.castEvent.timestamp - this.beginCastEvent.timestamp;
    const isInstantCast = channelingTime < CAST_BUFFER;
```

Operator preference is not always obvious, and right now you risk `this.beginCastEvent.timestamp < CAST_BUFFER` being executed before the - expression. It's safer to either use braces, or, better in this case, separate it into two variables. It's better because in this case it makes things more readable.